src/plugins/dataflow/components/dataflow-program.tsx

```diff
@@ -513,12 +520,24 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     this.programEditor.clear();
   };
 
+  private createCaseIdForTick = (collectedTime: number) => {
+    return collectedTime + "*" + this.props.tileId;
+  };
```

Comment:
1. `newCaseId()` is preferable to `collectedTime` here as it guarantees uniqueness regardless of sampling rate
2. Do you ever make use of the fact that the `tileId` is encoded into the case id? Did you have a particular use case in mind?

Depending on the answer to #2, you may not need to generate your own case id at all.

Reply:
@dennisrcao might have more insight on this, but I think we were thinking that a timestamp-based id would be good for ordering reasons - like verifying or restoring the order of cases?  Would it be better to add a straight ahead incrementing attribute? 

For the encoding, I believe we were anticipating there being a scenario in which we might want to be able to trace a case back to the tile that created it just by testing the id, but I suppose since the tile id is the `providerId` within the dataset that owns the case, that is not necessary.

Reply:
>I think we were thinking that a timestamp-based id would be good for ordering reasons

Default-generated case ids as provided by the `newCaseId()` function (which uses `ulid` under the hood) already have this property along with additional protections to handle the case of generating multiple cases with the same time stamp.